spec/models/miq_request_workflow_spec.rb

```diff
@@ -512,7 +512,7 @@
   context "#folder_to_datacenter" do
     before do
       datacenter.ext_management_system = ems
-      attrs = datacenter.attributes.merge(:object => datacenter, :ems => ems)
+      attrs = datacenter.attributes.merge(:object => workflow.ems_folder_to_hash_struct(datacenter), :ems => ems)
```

Comment:
As discussed I think this is the best fix for now.  May need to revisit later so wanted to document that the `ems_xml_nodes` in the workflow should be working with hash_struct instances.  I believe this is the first occurrence of the hash_struct version of an object doing anything more than exposing a few of the properties of the active_record object.